File: app/src/androidTest/java/daotests/MentionDaoTest.java
package daotests;

import android.content.Context;

import com.klinker.android.twitter_l.data.roomdb.TalonDatabase;

import org.junit.After;
import org.junit.AfterClass;
import org.junit.Before;
import org.junit.BeforeClass;
import org.junit.Test;

import androidx.room.Room;
import androidx.test.platform.app.InstrumentationRegistry;

public class MentionDaoTest extends DaoTest {

    @BeforeClass
    public static void initDatabase() {
        Context context = InstrumentationRegistry.getInstrumentation().getTargetContext();
        testDatabase = Room.inMemoryDatabaseBuilder(context, TalonDatabase.class)
                .build();
    }

    @Test
    public void insertMentionTest() {

    }

    @Test
    public void insertMentionWithConflict() {

    }

    @Test
    public void deleteMention() {

    }

    @Test
    public void getLatestScreenName() {

    }

    @Test
    public void getLatestUsers() {


    }


    @Test
    public void trimDatabaseTrimsDatabase() {

    }

    @Test
    public void trimDatabaseDoesNotTrimSmallDatabase() {

    }

    @Test
    public void mentionsCascadeDelete() {

    }


    private String[] mockUser() {
        return null;
    }

    private String mockTweet() {
        return null;
    }


    @After
    public void clearTables() {
        testDatabase.clearAllTables();
    }

    @AfterClass
    public void closeDatabase() {
        testDatabase.close();
        testDatabase = null;
    }
}

File: app/src/androidTest/java/daotests/MentionDaoTest.kt
package daotests

import android.content.ContentValues
import android.content.Context
import android.content.SharedPreferences
import android.database.sqlite.SQLiteDatabase

import com.klinker.android.twitter_l.data.roomdb.TalonDatabase

import org.junit.After
import org.junit.AfterClass
import org.junit.Before
import org.junit.BeforeClass
import org.junit.Test

import androidx.room.Room
import androidx.test.platform.app.InstrumentationRegistry
import com.klinker.android.twitter_l.data.roomdb.daos.MentionDao
import com.klinker.android.twitter_l.data.roomdb.entities.Mention
import com.klinker.android.twitter_l.data.roomdb.entities.Tweet
import com.klinker.android.twitter_l.data.roomdb.entities.User
import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.Matchers.*

class MentionDaoTest : DaoTest()
